pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -410,8 +407,18 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
     def _share_information_to_prevent_deadlock(self):
         self._share_pids()
 
-        # remove `PL_DDP_SYNC_TMPDIR` from os.environ
-        self._sync_dir = os.environ.pop("PL_DDP_SYNC_TMPDIR", None)
+        # there should be a unique sync_dir per nodes.
+        if self.local_rank == 0:
+            # create a temporary directory used to synchronize processes on deadlock.
+            self._sync_dir = tempfile.mkdtemp()
+
+        sync_dirs = []
+        node_zero = 0
+        for _ in range(self.num_nodes):
+            sync_dirs.append(self.broadcast(self._sync_dir, node_zero))
+            node_zero += self.world_size // self.num_nodes
```

Comment:
noob q: what is this calculating?

also, wouldn't it equal

```suggestion
    node_zero = (self.world_size // self.num_nodes) * self.num_nodes
```

Reply:
It is getting the indices of each first rank of each node and performing a broadcast of its local sync_dir

Reply:
Example: 4 nodes with 8 gpus: [0, 8, 16, 24]

Reply:
@carmocca